core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -2634,6 +2651,10 @@ private List<CatalogFunctionMetadata> getFunctions(Session session, ConnectorMet
                 .map(function -> new CatalogFunctionMetadata(catalogHandle, name.getSchemaName(), function))
                 .forEach(functions::add);
 
+        scalarFunctionRegistry.listFunctions(catalogHandle, name.getSchemaName()).stream()
```

Comment:
Is it possible to provide all functions from a connector in a uniform way?
Here we have two separate mechanisms for the existing connector-provided functions and the new annotation-based connector-provided functions. 
It makes the code complicated, and It is brittle in that the precedence is important.
For example here in `MetadataManager` we first list the "traditional" functions, and then the "new" functions.
However in `FunctionManager.getScalarFunctionImplementationInternal`, we first check the "new" functions.
If there is a duplicate, it could lead to a mismatch.